io/io/v7/src/TFile.cxx

```diff
@@ -97,7 +97,7 @@ class TV6Storage: public ROOT::Experimental::Internal::TFileStorageInterface {
   }
 
   void WriteMemoryWithType(std::string_view name, const void* address, TClass* cl) final {
-    fOldFile->WriteObjectAny(address, cl, name.to_string().c_str());
+     fOldFile->WriteObjectAny(address, cl, name.data());
```

Comment:
Could you point me to the comment in the code which states that it's required? At first look I don't see it explicitly mentioned in `TBufferFile::WriteObjectClass` or `TBufferFile::WriteObjectAny` and I didn't see any breakage in CMSSW.